processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -451,7 +453,7 @@ public boolean isRollup()
   );
 
   // Note: This method needs to be thread safe.
-  protected abstract Integer addToFacts(
+  protected abstract Pair<Integer, List<String>> addToFacts(
```

Comment:
I think `Pair` makes code difficult to understand because I have to imagine what makes the pair. It would be much great to add a new class.

Reply:
Changed this to use a new class instead of Pair